meson.build

```diff
@@ -0,0 +1,376 @@
+        '-Wc2x-compat',
+        '-Wc2x-extensions',
+        '-Wpre-c2x-compat',
+        '-Warray-bounds-pointer-arithmetic',
+        '-Wassign-enum',
+        '-Wconditional-uninitialized',
+        '-Wdocumentation',
+        '-Wduplicate-enum',
+        '-Wempty-translation-unit',
+        '-Wflexible-array-extensions',
+        '-Wmissing-variable-declarations',
+        '-Wnewline-eof',
+        '-Wshift-sign-overflow',
+        '-Wstring-conversion',
+    ]
+
+    lzma_c_args += cc.get_supported_arguments(cflags_try)
+endif
+
+config_h.set10('HAVE_VISIBILITY', cc.has_function_attribute('visibility'),
+    description: 'Define to 1 if the compiler supports simple visibility declarations.'
+)
+
+ctor_fct_src = '''
+__attribute__((__constructor__))
+static void my_constructor_func(void) { return; }
+int main() { return 0; }
+'''
+if cc.compiles(ctor_fct_src, name : 'constructor attribute for functions')
+    config_h.set10('HAVE_FUNC_ATTRIBUTE_CONSTRUCTOR', true,
+        description: 'Define to 1 if __attribute__((__constructor__)) is supported for functions.'
+    )
+endif
+
+cc.has_type('uint32_t', prefix: '#include <stdint.h>')
+
+time_fcts = [
+    [ 'futimens',  [ '#include <sys/stat.h>' ] ],
+    [ 'futimes',   [ '#include <sys/time.h>' ] ],
+    [ 'futimesat', [ '#include <fcntl.h>', '#include <sys/time.h>' ] ],
+    [ 'utimes',    [ '#include <sys/time.h>' ] ],
+    [ '_futime',   [ '#include <sys/utime.h>' ] ],
+    [ 'utime',     [ '#include <sys/types.h>', '#include <utime.h>' ] ],
+]
+foreach f : time_fcts
+    if cc.has_function(f[0], prefix : f[1])
+        config_h.set10('HAVE_' + f[0].to_upper(), true,
+            description: 'Define to 1 if the ' + f[0] + '() function is available.'
+        )
+        break
+    endif
+endforeach
+
+if cc.has_function('posix_advice', prefix : '#include <fcntl.h>')
+    config_h.set10('HAVE_POSIX_ADVICE', true,
+        description: 'Define to 1 if the posix_advice() function is available.'
+    )
+endif
+
+### Filters
+
+## encoders
+encoder_msg = ''
+decoder_msg = ''
+
+supported_filters = [
+    'lzma1',
+    'lzma2',
+    'delta',
+    'x86',
+    'powerpc',
+    'ia64',
+    'arm',
+    'armthumb',
+    'arm64',
+    'sparc',
+    'riscv',
+]
+
+simple_filters = [
+    'x86',
+    'powerpc',
+    'ia64',
+    'arm',
+    'armthumb',
+    'arm64',
+    'sparc',
+    'riscv',
+]
+
+# FIXME foreach i in array; set_variable(f'have_@i@', true) endforeach
+
+foreach f : supported_filters
+    if get_option('encoders').contains(f)
+        config_h.set10('HAVE_ENCODER_' + f.underscorify().to_upper(), true,
+            description: 'Define to 1 if ' + f + ' encoder is enabled.'
+        )
+        encoder_msg += f + ' '
+    endif
+    if get_option('decoders').contains(f)
+        config_h.set10('HAVE_DECODER_' + f.underscorify().to_upper(), true,
+            description: 'Define to 1 if ' + f + ' decoder is enabled.'
+        )
+        decoder_msg += f + ' '
+    endif
+endforeach
```

Comment:
It looks like `supported_filters` is just duplicating the list of `encoders` and `decoders` defined in `meson_options.txt`, so we can simplify to:

```suggestion
simple_filters = [
    'x86',
    'powerpc',
    'ia64',
    'arm',
    'armthumb',
    'arm64',
    'sparc',
    'riscv',
]

# FIXME foreach i in array; set_variable(f'have_@i@', true) endforeach

foreach f : get_option('encoders')
    config_h.set10('HAVE_ENCODER_' + f.underscorify().to_upper(), true,
        description: 'Define to 1 if ' + f + ' encoder is enabled.'
    )
    encoder_msg += f + ' '
endforeach
foreach f : get_option('decoders')
    config_h.set10('HAVE_DECODER_' + f.underscorify().to_upper(), true,
        description: 'Define to 1 if ' + f + ' decoder is enabled.'
    )
    decoder_msg += f + ' '
endforeach
```

Reply:
fixed in 132cde1b031a0308a5408f7acca4ccaa9800dd36